plugins/action/win_jcliff.py

```diff
@@ -0,0 +1,94 @@
+        templates = self._loader.path_dwim_relative(self._loader.get_basedir(),
+                                                    'templates/rules', template_name)
+        if not os.path.isfile(templates):
+            templates = _get_role_home() + '/templates/rules/' + template_name
+
+        with open(templates, 'r') as file:
+            data = file.read()
+        templar = Templar(loader=self._loader, variables=subsystem_values)
+        return _write_template_result_to_file(templar.template(data))
+
+    def _deploy_custom_rules_if_any(self, tmp_remote_src):
+        if 'rule_file' in self._task.args:
+            custom_rulesdir = self._task.args['rule_file']
+            if custom_rulesdir is not None:
+                for custom_rule_file in os.listdir(custom_rulesdir):
+                    self._transfer_file(custom_rulesdir + "/" + custom_rule_file,
+                                        tmp_remote_src + custom_rule_file +
+                                        "-custom" + self.TARGET_FILENAME_SUFFIX)
+
+    def _build_and_deploy_jcliff_rule_files(self, tmp_remote_src):
+        template_name_by_subsys = {
+            'drivers': 'drivers.j2',
+            'datasources': 'datasource.j2',
+            'system_props': 'system-properties.j2',
+            'deployments': 'deployments.j2',
+            'logging': 'logging.j2',
+            'keycloak': 'keycloak.j2'
+        }
+        subsystems = self._task.args['subsystems']
+        if subsystems is not None:
+            for subsys in subsystems:
+                for key in subsys.keys():
+                    if key in ('drivers', 'datasources', 'keycloak'):
+                        for index, subsystem_values in enumerate(subsys[key]):
+                            self._transfer_file(
+                                self._template_from_jinja_to_yml(
+                                    template_name_by_subsys[key],
+                                    {"values": subsystem_values}),
+                                tmp_remote_src + key + "-" +
+                                str(index) + self.TARGET_FILENAME_SUFFIX)
+                    if key in ('system_props', 'deployments', 'logging'):
+                        self._transfer_file(self._template_from_jinja_to_yml(
+                            template_name_by_subsys[key], {"values": subsys[key]}),
+                            tmp_remote_src + key + self.TARGET_FILENAME_SUFFIX)
+
+    def run(self, tmp=None, task_vars=None):
+        tmp_remote_src = self._make_tmp_path()
+        self._build_and_deploy_jcliff_rule_files(tmp_remote_src)
+        self._deploy_custom_rules_if_any(tmp_remote_src)
+        result = super(ActionModule, self).run(tmp, task_vars)
+        new_module_args = self._task.args.copy()
+        new_module_args.update(dict(remote_rulesdir=tmp_remote_src,))
+        result.update(self._execute_module(
+            module_name='win_jcliff', module_args=new_module_args, task_vars=task_vars))
```

Comment:
@sabre1041 The module_name in this line is the only difference between the action/win_jcliff.py and the action/jcliff.py. We could use an if statement and a call platform.system() to determine whether or not we are running on Windows, which will make both file completely identical. 

Then, instead of duplicating the source file, we could simply ask the jcliff role to duplicate the file for us. This way we would reduce code duplication. Alternatively, we could just add a build script that duplicate the file before doing the release.

Ideally, duplicating the file should happen during the collection build phase ($ ansible-galaxy collection build), but looking at the source code of Ansible, this build phase seems pretty shallow and not really matching what we Java/JEE dev are used to in terms of build.

WDTK?

Reply:
I've set up a PR to update the action/jcliff.py according to my proposal: [PR29](https://github.com/wildfly-extras/ansible_collections_jcliff/pull/29), I also rework my previous "hack" that remove the powershell on Linux to [generate the actions/win_jcliff.py](https://github.com/wildfly-extras/ansible_collections_jcliff/pull/27).

All of this is far from perfect, but I think this approach is better than duplicated sources.